### PR TITLE
fixing peacekraber in sp ( again )

### PR DESCRIPTION
**Pull request by catornot**
the previous pr ( #382 ) didn't work for everyone. this time it should.
[https://canary.discord.com/channels/920776187884732556/987319483846778890/987459128878313513](https://canary.discord.com/channels/920776187884732556/987319483846778890/987459128878313513)

### CODE REVIEW

**Review by ASpoonPlaysGames — APPROVED** [review]
Whether or not this does fix the issue in SP or if there are more issues doesn't really matter, peacekraber should be enabled on both MP and SP, no reason not to.

**Review by EladNLG — APPROVED** [review]

**Review by GeckoEidechse — CHANGES_REQUESTED** [review]
Uh, so even with the PR I still cannot do `give mp_weapon_peacekraber` in Singleplayer campaign when I tested.

**Comment by catornot** [comment]
yeah because you need to precache it.

**Review by GeckoEidechse — APPROVED** [review]

**Comment by ASpoonPlaysGames** [comment]
Anything blocking this from being merged?

**Comment by uniboi** [comment]
@catornot update your branch

**Comment by catornot** [comment]
oh no I merged co-op

**Comment by catornot** [comment]
huh should have fixed idk

**Comment by uniboi** [comment]
you need to merge main into sp-fix2

**Comment by catornot** [comment]
😅

**Review by uniboi — APPROVED** [review]
image
works in both mp and sp, code is good

File: Northstar.Custom/mod.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
 		},
 		{
 			"Path": "weapons/mp_weapon_peacekraber.nut",
-			"RunOn": "( CLIENT || SERVER ) && MP"
+			"RunOn": "( CLIENT || SERVER )"
```
Comment (uniboi) [inline Northstar.Custom/mod.json:42]:
```suggestion
			"RunOn": "CLIENT || SERVER"
```

to keep style of the other context conditions

**Comment by GeckoEidechse** [comment]
> damn this is a lot of commit history for a single line change

And I cannot even test it atm cause I cannot launch SP without `Invalid Name` error xD

**Comment by uniboi** [comment]
> > damn this is a lot of commit history for a single line change
> 
> And I cannot even test it atm cause I cannot launch SP without `Invalid Name` error xD

allow insecure auth in customservers autoexec.cfg

**Comment by uniboi** [comment]
@GeckoEidechse did you manage to test yet?

**Review by ASpoonPlaysGames — CHANGES_REQUESTED** [review]
Technically speaking this works in testing, the peacekraber can be acquired in SP. However, `sh_northstar_custom_precache` is not ran on SP, which means that the peacekraber is not precached, meaning you cannot get it without setting `host_thread` to a value other than 1 (0 works) and then precaching it manually via console.

This should probably get fixed

**Comment by catornot** [comment]
It's supposed to be precached in co-op
[sh_coop_custom_precache.gnut](https://github.com/xamionex/NorthstarMods/blob/co-op/Northstar.Coop/mod/scripts/vscripts/sh_coop_custom_precache.gnut)

**Comment by ASpoonPlaysGames** [comment]
> 

sure, but that's not fixing the peacekraber in SP, that's fixing it in coop. The peacekraber remains in a broken state in normal SP

**Comment by ASpoonPlaysGames** [comment]
Btw from my own testing just now, just running `sh_northstar_custom_precache` on `CLIENT || SERVER` works just fine with no noticeable side effects in normal SP (triple threat works, as does the peacekraber)
If there are errors from running it with coop, then maybe remove the precaching on coop?

**Comment by catornot** [comment]
making `sh_northstar_custom_precache ` `CLIENT || SERVER` wasn't working for some people, weird stuff `¯\_(ツ)_/¯`

**Pull request closed without merging**